rust/src/ike/ikev1.rs

```diff
@@ -0,0 +1,158 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Frank Honza <frank.honza@dcso.de>
+
+use std;
+use crate::core::{STREAM_TOSERVER};
+use crate::log::*;
+use crate::applayer::*;
+use nom;
+use crate::ike::parser::*;
+use crate::ike::ike::{IKEState, IkeEvent};
+use std::collections::HashSet;
+
+#[derive(Default)]
+pub struct IkeV1Header {
+    pub maj_ver: Option<u8>,
+    pub min_ver: Option<u8>,
```

Comment:
Are these fields not a duplicate of the common header ?

Reply:
the existing implementation of ikev2 uses the data structures from an external lib (https://github.com/rusticata/ipsec-parser).
This implementation is bound to the IKEv2-Types.
I just didn't want to reimplement/change too much stuff there. Thats why data structures of both versions are more separated than needed. Elsewhise this could be a common Header....

Reply:
That would be nice if this `maj_ver` was not present in `IkeV1Header` and was present in some common `IkeHeader `, even if it is also redundant in IKEv2

Reply:
ok i'll add the common parts one layer above, found the redundancy initially indeed not so nice with ikev2.